### PR TITLE
fix: minor refinements to collection utils

### LINKED ISSUE #29535: Refine common collection utils
### Description

There is code duplication between the MultivaluedHashMap implementations. Also ConcurrentMultivaluedHashMap.addFirst is not concurrency safe https://github.com/keycloak/keycloak/blob/1d613d90378e486f661117e66840cacd37448107/common/src/main/java/org/keycloak/common/util/ConcurrentMultivaluedHashMap.java#L60

There are some slight improvements that can be made to CollectionUtil methods, and we can remove intersection.

### PR DESCRIPTION
**Pull request by shawkins**
closes: #29535

Not sure what the policy is on common method removal - we don't use it anywhere else, but if we assume extension developers could be using it, then I'll add a deprecation instead.

### CODE REVIEW

File: common/src/main/java/org/keycloak/common/util/CollectionUtil.java [hunk 1]
```diff
@@ -79,14 +66,4 @@ public static boolean isNotEmpty(Collection<?> collection) {
...
-    public static <T> Set<T> intersection(Collection<T> col1, Collection<T> col2) {
-        if (isEmpty(col1) || isEmpty(col2)) return Collections.emptySet();
-
-        final Collection<T> iteratorCollection = col1.size() <= col2.size() ? col1 : col2;
-        final Collection<T> searchCollection = iteratorCollection.equals(col1) ? col2 : col1;
-
-        return iteratorCollection.stream()
-                .filter(searchCollection::contains)
-                .collect(Collectors.toSet());
-    }
```
Comment (vmuzikar) [inline common/src/main/java/org/keycloak/common/util/CollectionUtil.java:91]: This is the only breaking change (if I haven't overlooked something). I would probably go ahead and just remove it. But I'd suggest adding a note to the upgrading guide (with a suggested code to use instead).

**Review by vmuzikar — CHANGES_REQUESTED** [review]
@shawkins Thanks for this enhancement. Code changes look good to me. Only thing we're missing from my perspective is some note in the upgrading guide.

I believe this PR also needs an approve from the @keycloak/core.

**Comment by shawkins** [comment]
@vmuzikar ok, added a note

**Review by vmuzikar — DISMISSED** [review]
LGTM, thank you.

Let's give the Core Team a chance to chime in as well.

**Review by vmuzikar — APPROVED** [review]

**Review by graziang — APPROVED** [review]
@shawkins Thanks, LGTM!

**Pull request merged**